connect_four.py

```diff
@@ -42,25 +42,48 @@ def __vertical_win(self, row: int, col: int, color: str) -> bool:
                         return True
                 else:
                     return False
-        
-        return False  #this return shouldn't be necessary
 
     def __horizontal_win(self, row: int, col: int, color: str) -> bool:
-        #TODO change logic to be # of same color on left + # of same color on right
-        adjacent_count = 0
-        for col in range(self.num_cols): 
-            if self.current[row][col] == color:
-                adjacent_count += 1
-                if adjacent_count >= self.win_req:
-                    return True
-            else:
-                    adjacent_count = 0
-        
-        return False
+
+        left_count, right_count = 0, 0
+
+        while col - 1 - left_count >= 0 and self.current[row][col - 1 - left_count] == color: #while there is a column to the left and that piece is of same color
```

Comment:
really nice code! love the definition of left_count, right_count = 0, 0 on the same line. aslo the return statement is incredibly clean and concise. good stuff